sodasql/scan/scan.py

```diff
@@ -440,12 +440,17 @@ def _run_sql_metric_default_and_run_tests(self, sql_metric: SqlMetricYml, resolv
 
         test_variables = self._get_test_variables(column_name_lower)
 
+        column_name = None
+        if column_name_lower:
+            scan_column = self.scan_columns[column_name_lower]
+            column_name = scan_column.column_name
```

Comment:
This looks like it is the right fix indeed.  2 comments:

1) There is a method below `_run_sql_metric_with_groups_and_run_tests` that would have to be updated in similar style/

2) I think it's better to pass the case sensitive column name or the scan_column as a parameter into the function.  If the scan column has both the column_name and the column_name_lower pecomputed than that might be best. But just passing the column_name (case sensitive) and recomputing the column_name_lower is good too.  In any case you will then have to revisit the places where the method is called.